projects/packages/jetpack-mu-wpcom/src/features/wpcom-navigation-export-filter/class-export-filter.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * WPCOM Navigation Export Filter
+ *
+ * Filters wp_navigation posts from WXR exports when the post_author
+ * is not a valid user on the current site.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+declare( strict_types = 1 );
+
+namespace Automattic\Jetpack\Jetpack_Mu_Wpcom\Wpcom_Navigation_Export_Filter;
+
+/**
+ * Export Filter class to handle wp_navigation post filtering during WordPress exports.
+ */
+class Export_Filter {
+	/**
+	 * Tracks whether an export is currently in progress.
+	 *
+	 * @var bool
+	 */
+	private $is_exporting = false;
+
+	/**
+	 * Constructor to set up the export filter.
+	 */
+	public function __construct() {
+		add_action( 'export_wp', array( $this, 'start_export_filtering' ), 5 );
+	}
+
+	/**
+	 * Starts filtering export queries by hooking into the query filter.
+	 *
+	 * @return void
+	 */
+	public function start_export_filtering(): void {
+		$this->is_exporting = true;
+		add_filter( 'query', array( $this, 'filter_export_queries' ) );
+	}
+
+	/**
+	 * Stops filtering export queries by removing the query filter.
+	 *
+	 * @return void
+	 */
+	public function stop_export_filtering(): void {
```

Comment:
I don't understand why we need the `is_exporting` state. From what I can see, the code that runs in prod (not test) never changes this state after initialization, and the state is always going to be `is_exporting = true`. Am I missing something here?